pkg/trivy/wrapper.go

```diff
@@ -1,68 +1,97 @@
 package trivy
 
 import (
-	"bytes"
-	"encoding/json"
 	"fmt"
 	"github.com/aquasecurity/harbor-scanner-trivy/pkg/etc"
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/ext"
 	log "github.com/sirupsen/logrus"
 	"golang.org/x/xerrors"
-	"io"
-	"io/ioutil"
-	"os"
 	"os/exec"
 	"strings"
 )
 
+const (
+	trivyCmd = "trivy"
+)
+
+type ImageRef struct {
+	Name     string
+	Auth     RegistryAuth
+	Insecure bool
+}
+
 // RegistryAuth wraps registry credentials.
 type RegistryAuth struct {
 	Username string
 	Password string
 }
 
 type Wrapper interface {
-	Run(imageRef string, auth RegistryAuth, insecureRegistry bool) (ScanReport, error)
+	Scan(imageRef ImageRef) (ScanReport, error)
 }
 
 type wrapper struct {
-	config etc.Trivy
+	config     etc.Trivy
+	ambassador ext.Ambassador
 }
 
-func NewWrapper(config etc.Trivy) Wrapper {
+func NewWrapper(config etc.Trivy, ambassador ext.Ambassador) Wrapper {
 	return &wrapper{
-		config: config,
+		config:     config,
+		ambassador: ambassador,
 	}
 }
 
-func (w *wrapper) Run(imageRef string, auth RegistryAuth, insecureRegistry bool) (report ScanReport, err error) {
-	log.WithField("image_ref", imageRef).Debug("Started scanning")
-
-	executable, err := exec.LookPath("trivy")
-	if err != nil {
-		return report, err
-	}
+func (w *wrapper) Scan(imageRef ImageRef) (report ScanReport, err error) {
+	log.WithField("image_ref", imageRef.Name).Debug("Started scanning")
 
-	reportFile, err := ioutil.TempFile(w.config.ReportsDir, "scan_report_*.json")
+	reportFile, err := w.ambassador.TempFile(w.config.ReportsDir, "scan_report_*.json")
 	if err != nil {
 		return report, err
```

Comment:
Since you already have named returns, you can simply do 
```suggestion
		return 
```

Reply:
Done